src/registry.rs

```diff
@@ -389,3 +391,51 @@ pub trait Metric: crate::encoding::EncodeMetric + Send + Sync + std::fmt::Debug
 
 impl<T> Metric for T where T: crate::encoding::EncodeMetric + Send + Sync + std::fmt::Debug + 'static
 {}
+
+pub trait Register {
```

Comment:
```suggestion
pub trait Registrant {
```

I guess something to be registered would be a registrant. What do you think? Is that more or less intuitive? I am not a native speaker.